src/resolve/resolved-dnshttps-openssl.c

```diff
@@ -0,0 +1,250 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#if !ENABLE_DNS_OVER_HTTPS
+#error This source file requires DNS-over-HTTPS to be enabled and OpenSSL to be available.
+#endif
+
+#include "resolved-dns-stream.h"
+#include "resolved-dnshttps.h"
+#include "resolved-manager.h"
+#include "hexdecoct.h"
+#include "build.h"
+#include "llhttp.h"
+#include "string.h"
+
+
+static char *dnshttps_current_header_field = NULL;
+static Hashmap *dnshttps_parser_data = NULL;
+
+
+static int on_body_content(llhttp_t *parser, const char *at, size_t length) {
+        int r;
+        http_header *entry;
+        entry = (http_header*)malloc(sizeof(http_header));
+        entry->at = at;
+        entry->len = length;
+
+        HeaderFields enum_field;
+        enum_field = BODY;
+
+        r = hashmap_put(dnshttps_parser_data, UINT_TO_PTR(enum_field), entry);
+        if (r < 0){
+                log_debug_errno(r, "Failed to put body into HTTP response hashmap.");
+                return r;
+        }
+
+        return 0;
+}
+
+static int on_header_field(llhttp_t *parser, const char *at, size_t length) {
+        /* store in current header global */
+
+        /* Allocate memory for dnshttps_current_header_field and add one extra byte for the null terminator */
+        dnshttps_current_header_field = malloc(length + 1);
+        if (dnshttps_current_header_field == NULL)
```

Comment:
we typically do non-NULL checks relying on C's downgrade-to-bool logic, rather than explicit NULL comparisons. in other words:

```c
if (!dnshttps_current_header_field)
```